jetty-util/src/main/java/org/eclipse/jetty/util/DateCache.java

```diff
@@ -152,89 +112,89 @@ public TimeZone getTimeZone()
+     * If it happens to be in the same second as the last
+     * formatNow call, then the format is reused.
      *
-     * @param inDate the date in milliseconds since unix epoch
-     * @return Formatted date
+     * @param inDate the date in milliseconds since unix epoch.
+     * @return Formatted date.
      */
     public String format(long inDate)
     {
-        long seconds = inDate / 1000;
-
-        Tick tick = _tick;
+        return formatTick(inDate)._string;
+    }
 
-        // Is this the cached time
-        if (tick == null || seconds != tick._seconds)
-        {
-            // It's a cache miss
-            return ZonedDateTime.ofInstant(Instant.ofEpochMilli(inDate), _zoneId).format(_tzFormat);
-        }
+    /**
+     * Format a date according to our stored formatter without using the cache.
+     *
+     * @param inDate the Date.
+     * @return Formatted date.
+     */
+    public String formatWithoutCache(Date inDate)
+    {
+        return _tzFormat.format(inDate.toInstant());
+    }
 
-        return tick._string;
+    /**
+     * Format a date according to our stored formatter without using the cache.
+     *
+     * @param inDate the date in milliseconds since unix epoch.
+     * @return Formatted date.
+     */
+    public String formatWithoutCache(long inDate)
+    {
+        return _tzFormat.format(Instant.ofEpochMilli(inDate));
     }
 
     /**
      * Format a date according to our stored formatter.
      * The passed time is expected to be close to the current time, so it is
      * compared to the last value passed and if it is within the same second,
-     * the format is reused.  Otherwise a new cached format is created.
+     * the format is reused. Otherwise, a new cached format is created.
      *
      * @param now the milliseconds since unix epoch
      * @return Formatted date
+     * @deprecated use {@link #format(long)}
      */
+    @Deprecated
     public String formatNow(long now)
     {
-        long seconds = now / 1000;
-
-        Tick tick = _tick;
-
-        // Is this the cached time
-        if (tick != null && tick._seconds == seconds)
-            return tick._string;
-        return formatTick(now)._string;
+        return format(now);
     }
 
+    @Deprecated
     public String now()
     {
         return formatNow(System.currentTimeMillis());
     }
 
+    @Deprecated
     public Tick tick()
     {
         return formatTick(System.currentTimeMillis());
     }
 
-    protected Tick formatTick(long now)
+    protected Tick formatTick(long inDate)
     {
-        long seconds = now / 1000;
+        long seconds = inDate / 1000;
 
         Tick tick = _tick;
         // recheck the tick, to save multiple formats
         if (tick == null || tick._seconds != seconds)
         {
-            String s = ZonedDateTime.ofInstant(Instant.ofEpochMilli(now), _zoneId).format(_tzFormat);
+            String s = formatWithoutCache(inDate);
             _tick = new Tick(seconds, s);
```

Comment:
This is now being used to format arbitrary times, so the tick cache could be hoping all over time.